README.md

```diff
@@ -0,0 +1,122 @@
+# EVM Toolkit (ETK)
+
+[![license](https://img.shields.io/badge/license-MIT%2FApache--2.0-blue)](https://github.com/quilt/etk)
+[![ci status](https://github.com/lightclient/eipv/workflows/ci/badge.svg)](https://github.com/quilt/etk/actions)
+
+ETK is a collection of tools for writing, reading, and analyzing EVM bytecode.
+
+## Tools
+
+* Assembler (eas)
+* Disassembler (disease)
+
+## Installation
+
+```console
+git clone git@github.com:quilt/etk.git
+cd etk
+cargo install --path etk-asm --features cli
+cargo install --path etk-analyze --features cli
+```
+
+## Usage
+
+```console
+$ eas contract.asm out.bin
+$ disease --bin-file out.bin
+    0:  CALLDATASIZE
+    1:  PUSH1 0x01
+    2:  DUP1
+    ...
+```
+
+## Assembly Format
+
+### Opcodes
+
+Opcodes can be used via their [mnemonic](etk-asm/src/parse/asm.pest).
+
+### Immediates
+
+Some opcodes accept immediate operands. The only EVM instruction that suports
+immediates natively is the `PUSH` family. ETK also allows for `jumpdest`
+instructions to take an immediate label.
+
+```asm
+; push a constant
+push1 1
+
+; push a label
+push4 .label
+
+; label a jumpdest
+jumpdest .label
+```
+
+### Expanders
+
+Expanders are built-in macros for common use cases. For example, selector
+generation is a requirement for being ABI compatible.
+
+```asm
+; push the first four bytes of `keccak("transfer(address, uint256)")` to the
+; stack.
+push4 selector("transfer(address, uint256)")
+```
+
+### Macros
+
+Coming soon!
+
+## Example Program
+
+```asm
+; -- Instructions --            -- Current stack layout --
+
+; Read the calldata into memory.
+calldatasize                    ; [calldatasize]
+push1 0                         ; [calldatasize, calldata_ptr]
+dup1                            ; [calldatasize, calldata_ptr, mem_ptr]
+calldatacopy                    ; []
+
+; Extract only the function selector
+push1 0                         ; [selector_ptr]
+mload                           ; [dirty_selector]
+push1 e0                        ; [dirty_selector, 28*8]
```

Comment:
Would you mind updating the docs with the `0x` prefix?

Reply:
Accidentally pushed that commit to this PR :) I updated the doc anyways.